src/staking/manager.ts

```diff
@@ -0,0 +1,834 @@
+    stakingTx: Transaction,
+    feeRate: number,
+  ): Promise<Transaction> {
+    const params = getBabylonParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!params) {
+      throw new Error(`Unable to find staking params for version ${stakingParamVersion}`);
+    }
+
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while creating signed unbonding transaction");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      params,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const { psbt } = staking.createWithdrawStakingExpiredPsbt(
+      stakingTx,
+      feeRate,
+    );
+
+    const signedWithdrawalPsbtHex = await this.btcProvider.signPsbt(
+      psbt.toHex()
+    );
+    return Psbt.fromHex(signedWithdrawalPsbtHex).extractTransaction();
+  }
+
+  /**
+   * Creates a signed withdrawal transaction for the expired slashing output that 
+   * ready to be sent to the BTC network.
+   * @param stakingInput - The staking inputs.
+   * @param stakingParamVersion - The param version that was used to create the EOI
+   * @param slashingTx - The slashing transaction.
+   * @param feeRate - The fee rate in satoshis per byte.
+   * @returns The signed withdrawal transaction.
+   */
+  async createSignedWithdrawSlashingBtcTransaction(
+    stakingInput: StakingInputs,
+    stakingParamVersion: number,
+    slashingTx: Transaction,
+    feeRate: number,
+  ): Promise<Transaction> {
+    const params = getBabylonParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!params) {
+      throw new Error(`Unable to find staking params for version ${stakingParamVersion}`);
+    }
+
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while creating withdraw slashing transaction");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      params,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const { psbt } = staking.createWithdrawSlashingPsbt(
+      slashingTx,
+      feeRate,
+    );
+
+    const signedSlashingPsbtHex = await this.btcProvider.signPsbt(
+      psbt.toHex()
+    );
+    return Psbt.fromHex(signedSlashingPsbtHex).extractTransaction();
+  }
+
+  /**
+   * Creates a proof of possession for the staker based on ECDSA signature.
+   * @param bech32Address - The staker's bech32 address.
+   * @returns The proof of possession.
+   */
+  async createProofOfPossession(
+    bech32Address: string,
+  ): Promise<ProofOfPossessionBTC> {
+    if (!this.btcProvider.signMessage) {
+      throw new Error("Sign message function not found");
+    }
+    // Create Proof of Possession
+    const bech32AddressHex = uint8ArrayToHex(fromBech32(bech32Address).data);
+    this.emit(
```

Comment:
I don't understand these events. Some times we emit them, others we don't. Is there an explainer somewhere?